### PR TITLE
Add loading spinner and multiple loading steps to setup-site flow

### PR DESCRIPTION
**Pull request by roo2**
change the loading progress bar to a spinner animation and add more loading steps. This handles the case of very long loading times better.

#### Testing instructions 
test the hero flow with:
`/start/domains?flags=signup/hero-flow`

### CODE REVIEW

File: client/signup/reskinned-processing-screen/index.jsx [hunk 1]
```diff
@@ -17,7 +18,19 @@ const useSteps = ( { flowName, hasPaidDomain, hasAppliedDesign } ) => {
 			steps = [ __( 'Your site will be live shortly.' ) ]; // copy from 'packages/launch/src/focused-launch/success'
 			break;
 		case 'setup-site':
-			steps = [ __( 'Applying design' ) ];
+			steps = [
+				__( 'Personalizing your site' ),
```
Comment (a8ci18n) [inline client/signup/reskinned-processing-screen/index.jsx:22]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'Start Personalizing Your Site Now' )` *ES Score: 8*

File: client/signup/reskinned-processing-screen/index.jsx [hunk 2]
```diff
@@ -34,10 +48,14 @@ const useSteps = ( { flowName, hasPaidDomain, hasAppliedDesign } ) => {
...
 export default function ReskinnedProcessingScreen( props ) {
 	const { __ } = useI18n();
+	const setupSiteFeatureEnabled = isEnabled( 'signup/setup-site-after-checkout' );
 
 	const steps = useSteps( props );
 	const totalSteps = steps.current.length;
-
+	const isSetupSite =
+		props.flowName === 'setup-site' ||
+		( props.flowName === 'onboarding' && setupSiteFeatureEnabled );
```
Comment (p-jackson) [inline client/signup/reskinned-processing-screen/index.jsx]: Unfortunately, there's a bunch of flows that might transition to the `setup-setup` flow. It's a bit of a hack to mention them by name, but maybe it's the best we can do for now.
Ideally, I think this screen would look up what the actual signup destination is to decide whether we're going to the `setup-site` flow, but alas, the destination isn't calculated until the very last minute.

```suggestion
	const flowsThatMightGoToSiteSetup = [ 'business', 'premium', 'personal', 'free', 'onboarding', 'onboarding-with-email', 'onboarding-registrationless', 'business-monthly', 'premium-monthly', 'personal-monthly', 'do-it-for-me' ];
	const isSetupSite =
		props.flowName === 'setup-site' ||
		( flowsThatMightGoToSiteSetup.includes( props.flowName ) && setupSiteFeatureEnabled );
```

File: client/signup/reskinned-processing-screen/index.jsx [hunk 3]
```diff
@@ -34,10 +48,14 @@ const useSteps = ( { flowName, hasPaidDomain, hasAppliedDesign } ) => {
 // to work with the onboarding signup flow.
 export default function ReskinnedProcessingScreen( props ) {
 	const { __ } = useI18n();
+	const setupSiteFeatureEnabled = isEnabled( 'signup/setup-site-after-checkout' );
```
Comment (p-jackson) [inline client/signup/reskinned-processing-screen/index.jsx]: ?

```suggestion
	const setupSiteFeatureEnabled = isEnabled( 'signup/setup-site-after-checkout' ) || isEnabled( 'signup/hero-flow' );
```

File: client/signup/reskinned-processing-screen/style.scss [hunk 4]
```diff
@@ -29,19 +29,77 @@ $progress-duration: 800ms;
 		}
 	}
 
+	&__progress-numbered-steps {
+		margin-top: 0.7em;
+		padding: 1em;
+		text-align: center;
+		color: var( --studio-gray-40 );
```
Comment (p-jackson) [inline client/signup/reskinned-processing-screen/style.scss:36]: 👍

**Review by p-jackson — COMMENTED** [review]
We could also hoist the CSS out into a more global location (e.g. `_loading.scss`) and change the classname to `wpcom-loading-elipsis` or something. That it's available everywhere and the server-rendered loading symbol will be able to share that class.

Edit: or we could just do the hoisting in the follow up PR

File: client/signup/reskinned-processing-screen/index.jsx (hunk 2 above)
Comment (roo2) [reply client/signup/reskinned-processing-screen/index.jsx]: Oh I think we do actually have access to the ultimate destination, and in fact it's being accessed already in the parent component, and used to determine the `hasDomainApplied` prop! I've updated this PR to pass in a flag to represent this. So now, using this flag, we will not need to test the feature flags or flow names directly; as they will have already been checked :)

Comment (p-jackson) [reply client/signup/reskinned-processing-screen/index.jsx]: Sweet!

File: client/signup/main.jsx [hunk 5]
```diff
@@ -587,9 +587,7 @@ class Signup extends Component {
 				<ReskinnedProcessingScreen
 					flowName={ this.props.flowName }
 					hasPaidDomain={ hasPaidDomain }
-					// If destination is not setup-site flow, we'll apply default design now
-					// because the user cannot choose design in current flow
-					hasAppliedDesign={ ! destination.startsWith( '/start/setup-site' ) }
+					isSetupSiteFlow={ destination.startsWith( '/start/setup-site' ) }
```
Comment (p-jackson) [inline client/signup/main.jsx]: I think this prop name is a bit confusing. Should it be like `isGoingToSetupSiteFlow`, or `isDestinationSetupSiteFlow`?

File: client/signup/reskinned-processing-screen/index.jsx [hunk 6]
```diff
@@ -17,13 +18,27 @@ const useSteps = ( { flowName, hasPaidDomain, hasAppliedDesign } ) => {
...
+				__( 'Adding a domain' ),
+				__( 'Applying a plan' ),
+				__( 'Securing your information' ),
+				__( 'Optimizing your content' ),
+				__( 'Closing the loop' ),
+			];
 			break;
 		default:
 			steps = [
 				__( 'Building your site' ),
```
Comment (p-jackson) [inline client/signup/reskinned-processing-screen/index.jsx]: A positive of removing the "Building your site" text as Ollie suggested here p1634278618340100/1634086694.194100-slack-C029SB8JT8S is that it'll centre the animation, making it easier to line up with the server-side loading animation.

File: client/signup/reskinned-processing-screen/index.jsx [hunk 7]
```diff
@@ -64,27 +81,41 @@ export default function ReskinnedProcessingScreen( props ) {
...
-			<p className="reskinned-processing-screen__progress-numbered-steps">
-				{
-					// translators: these are progress steps. Eg: step 1 of 4.
-					sprintf( __( 'Step %(currentStep)d of %(totalSteps)d' ), {
-						currentStep: currentStep + 1,
-						totalSteps,
-					} )
-				}
-			</p>
+			{ isSetupSiteFlow && (
```
Comment (p-jackson) [inline client/signup/reskinned-processing-screen/index.jsx]: In your animation p1634277774338900/1634086694.194100-slack-C029SB8JT8S your showing the ellipses during headstart. Should this logic instead be 
```suggestion
			{ ( isSetupSiteFlow || flowName === 'setup-site' )&& (
```

Another reason I thought the `isSetupSiteFlow` prop name was confusing 😃

File: client/signup/reskinned-processing-screen/index.jsx (hunk 6 above)
Comment (p-jackson) [reply client/signup/reskinned-processing-screen/index.jsx]: Ya, we should just remove all text when we're transitioning to the setup site flow p1634521893363800/1634086694.194100-slack-C029SB8JT8S

File: client/signup/reskinned-processing-screen/index.jsx (hunk 7 above)
Comment (roo2) [reply client/signup/reskinned-processing-screen/index.jsx]: oh yes! good catch, I confused myself with that 😅

**Review by p-jackson — APPROVED** [review]
I think we could make some improvements to the alignment of the ellipsis animation. But I'll merge now and continue to work on it as part of the server-side rendering changes.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6844519

Hi @roo2, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Personalizing your site`
- `Applying your theme`
- `Turning on hosting`
- `Enabling SSL encryption`
- `Switching on email`
- `Applying Jetpack essentials`
- `Adding a domain`
- `Applying a plan`
- `Securing your information`
- `Optimizing your content`
- `Closing the loop`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6844519).

**Pull request merged**